controllers/component_build_controller_pac.go

```diff
@@ -148,6 +156,162 @@ func (r *ComponentBuildReconciler) ProvisionPaCForComponent(ctx context.Context,
+	log := ctrllog.FromContext(ctx).WithName("TriggerPaCBuild")
+	ctx = ctrllog.IntoContext(ctx, log)
+
+	incomingSecret, incomingSecretName, reconcileRequired, err := r.ensureIncomingSecret(ctx, component)
+	if err != nil {
+		return false, err
+	}
+
+	repository, err := r.findPaCRepositoryForComponent(ctx, component)
+	if err != nil {
+		return false, err
+	}
+
+	if repository == nil {
+		return false, fmt.Errorf("PaC repository not found for component %s", component.Name)
+	}
+
+	repoUrl := component.Spec.Source.GitSource.URL
+	gitProvider, err := gitops.GetGitProvider(*component)
+	if err != nil {
+		log.Error(err, "error detecting git provider")
+		// There is no point to continue if git provider is not known.
+		return false, boerrors.NewBuildOpError(boerrors.EUnknownGitProvider, err)
+	}
+
+	pacSecret, err := r.ensurePaCSecret(ctx, component, gitProvider)
+	if err != nil {
+		return false, err
+	}
+
+	gitClient, err := gitproviderfactory.CreateGitClient(gitproviderfactory.GitClientConfig{
+		PacSecretData:             pacSecret.Data,
+		GitProvider:               gitProvider,
+		RepoUrl:                   repoUrl,
+		IsAppInstallationExpected: true,
+	})
+	if err != nil {
+		return false, err
+	}
+
+	// get target branch for incoming hook
+	targetBranch := component.Spec.Source.GitSource.Revision
+	if targetBranch == "" {
+		targetBranch, err = gitClient.GetDefaultBranch(repoUrl)
+		if err != nil {
+			return false, err
+		}
+	}
+
+	if repository != nil {
+		incomingUpdated := updateIncoming(repository, incomingSecretName, pacIncomingSecretKey, targetBranch)
+
+		if incomingUpdated {
+			if err := r.Client.Update(ctx, repository); err != nil {
+				log.Error(err, "failed to update PaC repository with incomings", "PaCRepositoryName", repository.Name)
+				return false, err
+			}
+			log.Info("Added incomings to the PaC repository", "PaCRepositoryName", repository.Name, l.Action, l.ActionUpdate)
+
+			// reconcile to be sure that Repository is updated
+			return true, nil
+		}
+	}
+
+	// reconcile to be sure that Secret is created
+	if reconcileRequired {
+		return true, nil
+	}
+
+	webhookTargetUrl, err := r.getPaCWebhookTargetUrl(ctx)
+	if err != nil {
+		return false, err
+	}
+
+	secretValue := string(incomingSecret.Data[pacIncomingSecretKey][:])
+
+	pipelineRunName := component.Name + pipelineRunOnPushSuffix
+
+	triggerURL := fmt.Sprintf("%s/incoming?secret=%s&repository=%s&branch=%s&pipelinerun=%s", webhookTargetUrl, secretValue, repository.Name, targetBranch, pipelineRunName)
+	HttpClient := GetHttpClientMethod()
+
+	resp, err := HttpClient.Post(triggerURL, "application/json", nil)
+	if err != nil {
+		return false, err
+	}
+
+	if resp.StatusCode != 200 && resp.StatusCode != 202 {
+		return false, fmt.Errorf("PaC incoming endpoint returned HTTP %d", resp.StatusCode)
+	}
+
+	log.Info(fmt.Sprintf("PaC build manually triggered push pipeline for component: %s", component.Name))
+	return false, nil
+}
```

Comment:
Would be nice to make a PaC API under `pkg/` like the `pkg/git/`.

Reply:
Not sure we need it for one request function.